src/ng/directive/ngOptions.js

```diff
@@ -269,8 +286,11 @@ var ngOptionsDirective = ['$compile', '$parse', function($compile, $parse) {
           var locals = getLocals(values[key], key);
           var label = displayFn(scope, locals);
           var selectValue = getTrackByValue(values[key], locals);
+          var disableWhen = disableWhenFn(scope, locals);
           watchedArray.push(selectValue);
-          watchedArray.push(label);
+          // for optimization, do not watch label or disableWhen if they weren't defined
+          if (match[2]) watchedArray.push(label);
+          if (match[4]) watchedArray.push(disableWhen);
```

Comment:
We can do more here. We don't need to even compute the `label` or `displayWhen` values if they are not specified. See https://github.com/angular/angular.js/pull/11052/files